### PR TITLE
[Team Cymru] Support Proxy

### PR DESCRIPTION
**Pull request by michal-dagan**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/XSUP-22078)

## Description
 Added support for the *Proxy URL* integration parameter.

### CODE REVIEW

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 1]
```diff
@@ -19,36 +22,46 @@
...
+        demisto.debug("after settimeout before makefile")
+        self.file = self.socket.makefile("rw")
+        demisto.debug("after makefile")
 
-    :return: Dictionary contains the results of the lookup API call.
-    :rtype: Dict[str, str]
-    """
-    raw_result = client.lookup(ip)
-    return vars(raw_result) if raw_result else None
+    def lookup(self, ip: str) -> Optional[Dict[str, Any]]:
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:40]:
```suggestion
    def lookup(self, ip: str) -> dict[str, str] | None:
```

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 2]
```diff
@@ -19,36 +22,46 @@
...
+        """Perform lookups by ip address and return ASN, Country Code, and Network Owner.
 
+        :type ip: ``str``
+        :param ip: string to add in the dummy dict that is returned
 
-def team_cymru_bulk_whois(client: Client, bulk: List[str]) -> Optional[Dict[str, Any]]:
-    """Perform lookups by bulk of ip addresses,
-    returning a dictionary of ip -> record (ASN, Country Code, and Netblock Owner.)
+        :return: Dictionary contains the results of the lookup API call.
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:46]:
```suggestion
        :return: Dictionary contains the results of the lookup API call if succeeded, else None
```
```suggestion
        :return: Dictionary contains the results of the lookup API call.
```

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 3]
```diff
@@ -19,36 +22,46 @@
...
+        :rtype: Dict[str, str]
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:47]:
```suggestion
        :rtype: Dict[str, str] or None
```

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 4]
```diff
@@ -19,36 +22,46 @@
...
+        """
+        raw_result = super().lookup(ip)
+        return vars(raw_result) if raw_result else None
 
-    :type client: ``Client``
-    :param client: cymruwhois client to use
-    :type bulk: ``list``
-    :param bulk: list of ip addresses
+    def lookupmany_dict(self, bulk: List[str]) -> Optional[Dict[str, Any]]:
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:52]:
```suggestion
    def lookupmany_dict(self, bulk: list[str]) -> dict[str, Any] | None:
```

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 5]
```diff
@@ -127,7 +140,7 @@ def parse_file(file_path_res: dict[str, str], delimiter: str = ",") -> List[str]
     return bulk_list
 
 
-def parse_ips_list(client: Client, ips_list: list[str]) -> List[CommandResults]:
+def parse_ips_list(client: CymruClient, ips_list: list[str]) -> List[CommandResults]:
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:143]:
```suggestion
def parse_ips_list(client: CymruClient, ips_list: list[str]) -> list[CommandResults]:
```

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 6]
```diff
@@ -178,7 +192,7 @@ def test_module(client: Client) -> str:
     return message
 
 
-def ip_command(client: Client, args: Dict[str, Any]) -> List[CommandResults]:
+def ip_command(client: CymruClient, args: Dict[str, Any]) -> List[CommandResults]:
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:195]:
```suggestion
def ip_command(client: CymruClient, args: dict[str, Any]) -> list[CommandResults]:
```

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 7]
```diff
@@ -199,13 +213,13 @@ def ip_command(client: Client, args: Dict[str, Any]) -> List[CommandResults]:
...
     # Call the Client function and get the raw response
-    result = team_cymru_ip(client, ip[0])
+    result = client.lookup(ip[0])
     if result:
         command_results.append(parse_ip_result(ip[0], result))
     return command_results
 
 
-def cymru_bulk_whois_command(client: Client, args: Dict[str, Any]) -> List[CommandResults]:
+def cymru_bulk_whois_command(client: CymruClient, args: Dict[str, Any]) -> List[CommandResults]:
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:222]:
```suggestion
def cymru_bulk_whois_command(client: CymruClient, args: dict[str, Any]) -> list[CommandResults]:
```

File: Packs/TeamCymru/Integrations/TeamCymru/README.md [hunk 8]
```diff
@@ -9,9 +9,9 @@ This integration was integrated and tested with version 1.0 of TeamCymru
 
     | **Parameter** | **Required** |
     | --- | --- |
-    | Trust any certificate (not secure) | False |
     | Use system proxy settings | False |
-
+    | Proxy URL | Supports socks4/socks5/http connect proxies \(e.g. socks5h://host:1080\). | False |
```
Comment (ShirleyDenkberg) [inline Packs/TeamCymru/Integrations/TeamCymru/README.md:13]:
```suggestion
    | Proxy URL | Supports socks4/socks5/http connect proxies \(e.g., socks5h://host:1080\). | False |
```

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.yml [hunk 9]
```diff
@@ -11,6 +11,15 @@ configuration:
   name: proxy
   required: false
   type: 8
+  section: Connect
+  advanced: true
+- display: Proxy URL
+  additionalinfo: "Supports socks4/socks5/http connect proxies (e.g. socks5h://host:1080)."
```
Comment (ShirleyDenkberg) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.yml:17]:
```suggestion
  additionalinfo: "Supports socks4/socks5/http connect proxies (e.g., socks5h://host:1080)."
```

File: Packs/TeamCymru/ReleaseNotes/1_0_5.md [hunk 10]
```diff
@@ -0,0 +1,3 @@
+#### Integrations
+##### Team Cymru
+- Added support for the *Proxy URL* integration parameter.
```
Comment (ShirleyDenkberg) [inline Packs/TeamCymru/ReleaseNotes/1_0_5.md:3]:
```suggestion
Added support for the *Proxy URL* integration parameter.
```

**Comment by ShirleyDenkberg** [comment]
@bziser @ostolero Doc review completed.

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.yml [hunk 11]
```diff
@@ -11,6 +11,16 @@ configuration:
   name: proxy
   required: false
   type: 8
+  section: Connect
+  advanced: true
+  defaultvalue: 'False'
```
Comment (michal-dagan) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.yml:16]: defaultvalue: 'false'

**Pull request merged**